### PR TITLE
Add app shortcuts 

### PR DESCRIPTION
**Pull request by mueller-ma**
See #1153

### CODE REVIEW

File: mobile/src/main/java/org/openhab/habdroid/ui/MainActivity.java [hunk 1]
```diff
@@ -821,8 +856,9 @@ public boolean onPrepareOptionsMenu(Menu menu) {
         Log.d(TAG, "onPrepareOptionsMenu()");
         MenuItem voiceRecognitionItem = menu.findItem(R.id.mainmenu_voice_recognition);
         @ColorInt int iconColor = ContextCompat.getColor(this, R.color.light);
-        voiceRecognitionItem.setVisible(
-                mConnection != null && SpeechRecognizer.isRecognitionAvailable(this));
+        boolean visible = mConnection != null && SpeechRecognizer.isRecognitionAvailable(this);
+        voiceRecognitionItem.setVisible(visible);
+        manageVoiceRecognitionShortcut(visible);
```
[the remark this answers is not in the document]
Comment (mueller-ma) [reply mobile/src/main/java/org/openhab/habdroid/ui/MainActivity.java]: > voice recognition shortcut should be updated in onConnectionChanged

I disagree: The server has no influence whether the mic icon in the action bar is shown.

> habpanel shortcut should be updated in onConnectionChanged (to false if connection == null)

If the connection is null, I guess the best option is to look at the saved server props from the previous server.

[the remark this answers is not in the document]
Comment (maniac103) [reply mobile/src/main/java/org/openhab/habdroid/ui/MainActivity.java]: > I disagree: The server has no influence whether the mic icon in the action bar is shown.

It kinda has: if there's no connection it should be hidden ;-)

> If the connection is null, I guess the best option is to look at the saved server props from the previous server.

But if it's null you can't open Habpanel anyway?

Both things really are the same issue: what I want to say there is that if connection is null while the activity is shown that's a sign of a non-working connection, which in turn means we'd need to hide the shortcuts.

[the remark this answers is not in the document]
Comment (maniac103) [reply mobile/src/main/java/org/openhab/habdroid/ui/MainActivity.java]: That is a very good point, but can also be turned around: What if your local server has Habpanel and the remote server does not? You're at home (-> habpanel shortcut appears), leave the app open, leave home (-> connects to remote server) -> habpanel shortcut is still present, but doesn't work. Same applies to notifications page, obviously.

The differentiation between local and remote server is what makes this somewhat tricky. The only chance to do this properly is
- saving last properties of *both* local and remote server
- killing those properties when removing the server from settings
- after fetching properties, update voice and habpanel shortcuts with the superset of local and remote server flags

Notifications shortcut should probably be enabled once there is a cloud connection and removed once the remote server is cleared...not easy to do though; we abstract away the reason why no cloud connection exists :-/

[the remark this answers is not in the document]
Comment (maniac103) [reply mobile/src/main/java/org/openhab/habdroid/ui/MainActivity.java]: :+1: for the first two. One could probably remove it if no remote server is set; but I guess the likelihood of a user first setting and then removing a remote server is slim anyway ... so :+1: to that as well.

[the remark this answers is not in the document]
Comment (mueller-ma) [reply mobile/src/main/java/org/openhab/habdroid/ui/MainActivity.java]: Done :)

File: mobile/src/main/java/org/openhab/habdroid/ui/MainActivity.java [hunk 2]
```diff
@@ -355,6 +371,15 @@ private void processIntent(Intent intent) {
                 CloudMessagingHelper.onNotificationSelected(this, intent);
                 onNotificationSelected(intent);
                 break;
+            case ACTION_HABPANEL_SELECTED:
+                mShouldStartHabpanel = true;
+                openHabpanelIfNeeded();
+                break;
+            case ACTION_VOICE_RECOGNITION_SELECTED:
+                if (mServerProperties != null) {
+                    launchVoiceRecognition();
```
Comment (mueller-ma) [inline mobile/src/main/java/org/openhab/habdroid/ui/MainActivity.java]: This should be wrapped in a `launchVoiceRecognitionIfNeeded()` function and be called after the serverprops have been fetched.

**Pull request merged**